R/vars-select.R

```diff
@@ -311,6 +333,51 @@ ind_check <- function(x) {
   }
 }
 
+inds_check <- function(x, vars, incl, dups, unrenamed, unrenamed_vars) {
+  renamers <- names(incl)[!unrenamed]
+
+  # Below we check that variables are renamed to a unique name. But we
+  # also want to allow renaming existing duplicates, which we remove
+  # from the checking set here.
+  if (any(dups)) {
+    ok <- names(dups)[dups]
+    renamers <- renamers[!renamers %in% ok]
+  }
+
+  if (vctrs::vec_duplicate_any(renamers)) {
+    dups <- vctrs::vec_duplicate_detect(renamers)
+    dups <- vctrs::vec_unique(renamers[dups])
+
+    probs <- map_chr(dups, function(dup) {
+      cols <- vars[incl[names(incl) == dup]]
+      cols <- glue::backtick(cols)
+      cols <- glue::glue_collapse(cols, sep = ", ", last = " and ")
+      glue::glue("* Columns {cols} are being renamed to `{dup}`.")
+    })
+    msg <- paste_line(
+      "Must use different names when renaming columns.",
```

Comment:
```suggestion
      "Must use unique names when renaming columns.",
```

I think it's best to re-use existing name repair vocabulary and levels whenever we can.

Reply:
good point thanks!